### PR TITLE
Proj1700: Move XML node to a new line

### PR DESCRIPTION
**Pull request by CptWesley**
Documentation coming later.

### CODE REVIEW

**Comment by Corniel** [comment]
I'd like to implement #135 first. It might turn out that this rule is not longer needed then, but we it might still be a nice addition.

File: src/DotNetProjectFile.Analyzers/Analyzers/MsBuild/PlaceXmlNodesOnSeperateLines.cs [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+using System.Xml;
+
+namespace DotNetProjectFile.Analyzers.MsBuild;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+public sealed class PlaceXmlNodesOnSeperateLines() : MsBuildProjectFileAnalyzer(Rule.PlaceXmlNodesOnSeperateLines)
+{
+    protected override void Register(ProjectFileAnalysisContext context)
+    {
+        var nodes = GetFlatChildren(context.Project).ToArray();
```
Comment (Corniel) [inline src/DotNetProjectFile.Analyzers/Analyzers/MsBuild/PlaceXmlNodesOnSeperateLines.cs:11]:
```suggestion
        var nodes = context.Project.DescendantsAndSelf();
```

File: src/DotNetProjectFile.Analyzers/Analyzers/MsBuild/PlaceXmlNodesOnSeperateLines.cs [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+            yield return child;
+            foreach (var grandchild in GetFlatChildren(child))
+            {
+                yield return grandchild;
+            }
+        }
+    }
+
+#pragma warning disable S3776 // Max complexity of 15. Justification: Logic state machine relatively well contained.
+    private static (int StartLine, int EndLine) GetLineNumbers(Node node)
```
Comment (Corniel) [inline src/DotNetProjectFile.Analyzers/Analyzers/MsBuild/PlaceXmlNodesOnSeperateLines.cs:52]: Why is this needed? node.LineInfo.LinePositionSpan() gives you this already. And depth can be resolved with `node. AncestorsAndSelf().Count()`;

Comment (CptWesley) [reply src/DotNetProjectFile.Analyzers/Analyzers/MsBuild/PlaceXmlNodesOnSeperateLines.cs:52]: It only gives you the starting line, not the ending line

Comment (Corniel) [reply src/DotNetProjectFile.Analyzers/Analyzers/MsBuild/PlaceXmlNodesOnSeperateLines.cs:52]: It gives both, but the end line is wrong. I'd like to fix that differently. See #136

Comment (Corniel) [reply src/DotNetProjectFile.Analyzers/Analyzers/MsBuild/PlaceXmlNodesOnSeperateLines.cs:52]: I would be in favour of closing this one. #135 also covers this one.

Comment (CptWesley) [reply src/DotNetProjectFile.Analyzers/Analyzers/MsBuild/PlaceXmlNodesOnSeperateLines.cs:52]: I would like to give the changes in #135 a shot at my test case and adjust the expected results accordingly. If this indeed produces equivalent warnings, I would still like to explicitly add the test cases to the test suite (while removing all other logic currently present).

Comment (CptWesley) [reply src/DotNetProjectFile.Analyzers/Analyzers/MsBuild/PlaceXmlNodesOnSeperateLines.cs:52]: Seems to work indeed:

```
Verification failed: 4 extra, 4 missing.
[+] Proj1700 The element <TargetFramework> has not been properly indented. @[(2,17)-(2,33)]
[+] Proj1700 The element </ItemGroup> has not been properly indented. @[(6,74)-(6,86)]
[+] Proj1700 The element <PackageReference> has not been properly indented. @[(4,13)-(4,101)]
[+] Proj1700 The element </ItemGroup> has not been properly indented. @[(9,43)-(9,55)]
[-] Proj1700 Move XML node '<TargetFramework>net8.0</TargetFramework>' to a new line. @[(2,18)-(2,19)]
[-] Proj1700 Move XML node '<PackageReference Include="Microsoft.CodeAnalysis.Workspaces.Common" Version="4.10.0" />' to a new line. @[(4,14)-(4,101)]
[-] Proj1700 Move XML node '<PackageReference Include="Qowaiv.Analyzers.CSharp" Version="2.0.0" />' to a new line. @[(6,5)-(6,6)]
[-] Proj1700 Move XML node '<Compile Include="../common/Code.cs" />' to a new line. @[(9,5)-(9,6)]
```
